src/EnergyPlus/UnitarySystem.cc

```diff
@@ -11608,6 +11657,14 @@ namespace UnitarySystems {
                 LatentLoad = false;
             }
 
+            // disable waterside economizer if the condition is NOT favorable
+            if (this->m_waterSideEconomizerFlag) {
+                if (!this->runWaterSideEconomizer) {
+                    SensibleLoad = false;
+                    LatentLoad = false;
+                }
+            }
+
```

Comment:
if a UnitarySystem is used as a water-side economizer but the condition is not favorable for the UnitarySystem to operate, then this code sets these two local variables ("SensibleLoad " and "LatentLoad")  to FALSE as a means to turn off this UnitarySystem in controlCoolingSystemToSP() function.